platform/darwin/src/http_file_source.mm

```diff
@@ -229,6 +229,10 @@ void cancel() {
 std::unique_ptr<AsyncRequest> HTTPFileSource::request(const Resource& resource, Callback callback) {
     auto request = std::make_unique<HTTPRequest>(callback);
     auto shared = request->shared; // Explicit copy so that it also gets copied into the completion handler block below.
+    
+    if ([MGLNetworkConfiguration sharedManager].stopsRequests) {
+        return std::move(request);
```

Comment:
The problem with this implementation is when you resume, it will not re-request the previously requested (but not fulfilled) network requests. 

We have an API for this: `mbgl::NetworkStatus::Set(NetworkStatus::Status::Offline)`.